File: test/clojure/bench/box_churn.clj
;; Mirror of test/jacl/bench/box_churn.jacl.
;;
;; Allocates 2000 short-lived single-element atoms (Clojure's closest
;; analog to a JACL box - thread-local mutable container with deref/reset
;; semantics), dereferences each, accumulates the sum. Validates: 1999000.

(ns bench.box-churn)

(defn run []
  (loop [i 0 acc 0]
    (if (< i 2000)
      (let [b (atom i)]
        (recur (inc i) (+ acc @b)))
      acc)))

File: test/python/bench/box_churn.py
"""Mirror of test/jacl/bench/box_churn.jacl.

Allocates 2000 short-lived single-element lists (closest Python analog to
a JACL box), dereferences each, accumulates the sum. Validates: 1999000.
"""


def run() -> int:
    acc = 0
    for i in range(2000):
        b = [i]
        acc += b[0]
    return acc


if __name__ == "__main__":
    print(run())
